File: Scripts/results/generated_code_results/RN/Education/OnlineTeachingSoftware/NotificationScreen.js
import React from 'react';
import { View, Text, Image, TouchableOpacity, ScrollView, StyleSheet, StatusBar, Pressable } from 'react-native';
import { useNavigation } from '@react-navigation/native';
const Notifications = () => {
const navigation = useNavigation();
const notificationData = [
{
type: 'Purchase',
title: 'Purchase Completed!',
description: 'You already bought a class from Jerremy Mamika. Enjoy the class! 👋',
time: '2 m ago',
icon: require('../assets/snack-icon.png'),
},
{
type: 'Message',
title: 'Jerremy Send You a Message',
description: "Hi Magdalena, I've seen your challenge...",
action: 'Reply the message',
time: '2 m ago',
icon: require('../assets/snack-icon.png'),
},
{
type: 'Promotion',
title: 'Ramadhan Flash Sale!',
description: 'Get 20% discount for first transaction in this month!😍😍',
time: '2 m ago',
icon: require('../assets/snack-icon.png'),
},
{
type: 'Achievement',
title: 'Challange Completed',
description: 'Congratulation! you have completed course by Jerremy Mamika',
time: '10 m ago',
icon: require('../assets/snack-icon.png'),
},
];
return (
<View style={styles.container}>
<StatusBar barStyle="dark-content" />
<View style={styles.header}>
<Pressable onPress={() => navigation.goBack()} style={styles.backButton}>
<Image source={require('../assets/snack-icon.png')} style={styles.backIcon} />
</Pressable>
<Text style={styles.headerTitle}>Notification</Text>
<TouchableOpacity style={styles.settingsButton}>
<Image source={require('../assets/snack-icon.png')} style={styles.settingsIcon} />
</TouchableOpacity>
</View>
  <ScrollView style={styles.notificationList}>
    <Text style={styles.sectionTitle}>Recent</Text>
    {notificationData.map((notification, index) => (
      <View key={index} style={styles.notificationItem}>
        <Image source={notification.icon} style={styles.notificationIcon} />
        <View style={styles.notificationContent}>
          <Text style={styles.notificationTitle}>{notification.title}</Text>
          <Text style={styles.notificationDescription}>{notification.description}</Text>
          {notification.action && (
            <TouchableOpacity>
              <Text style={styles.actionText}>{notification.action}</Text>
            </TouchableOpacity>
          )}
        </View>
        <Text style={styles.notificationTime}>{notification.time}</Text>
      </View>
    ))}
  </ScrollView>
</View>
);
};
const styles = StyleSheet.create({
container: {
flex: 1,
backgroundColor: '#FFFFFF',
},
header: {
flexDirection: 'row',
justifyContent: 'space-between',
alignItems: 'center',
paddingHorizontal: 20,
paddingTop: 50,
paddingBottom: 20,
},
backButton: {
padding: 5,
},
backIcon: {
width: 24,
height: 24,
},
headerTitle: {
fontSize: 20,
fontWeight: 'bold',
},
settingsButton: {
padding: 5,
},
settingsIcon: {
width: 24,
height: 24,
},
notificationList: {
flex: 1,
},
sectionTitle: {
fontSize: 18,
fontWeight: 'bold',
marginLeft: 20,
marginBottom: 10,
},
notificationItem: {
flexDirection: 'row',
padding: 20,
borderBottomWidth: 1,
borderBottomColor: '#F0F0F0',
},
notificationIcon: {
width: 40,
height: 40,
borderRadius: 20,
marginRight: 15,
},
notificationContent: {
flex: 1,
},
notificationTitle: {
fontSize: 16,
fontWeight: 'bold',
marginBottom: 5,
},
notificationDescription: {
fontSize: 14,
color: '#666666',
marginBottom: 5,
},
actionText: {
color: '#6B4EFF',
fontSize: 14,
},
notificationTime: {
fontSize: 12,
color: '#999999',
},
});
export default Notifications;

File: Scripts/results/generated_code_results/Flutter/Education/OnlineTeachingSoftware/NotificationScreen.dart
import 'package:flutter/material.dart';



class NotificationsScreen extends StatelessWidget {

  @override

  Widget build(BuildContext context) {

    return Scaffold(

      appBar: AppBar(

        leading: IconButton(

          icon: Icon(Icons.arrow_back),

          onPressed: () => Navigator.pop(context),

        ),

        title: Text('Notification'),

        actions: [

          IconButton(

            icon: Icon(Icons.settings),

            onPressed: () {

              // TODO: Implement settings action

            },

          ),

        ],

      ),

      body: ListView(

        children: [

          Padding(

            padding: const EdgeInsets.all(16.0),

            child: Text(

              'Recent',

              style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),

            ),

          ),

          _buildNotificationItem(

            icon: Icons.shopping_cart,

            title: 'Purchase Completed!',

            description: 'You already bought a class from Jerremy Mamika. Enjoy the class! 👋',

            time: '2 m ago',

          ),

          _buildNotificationItem(

            icon: Icons.person,

            title: 'Jerremy Send You a Message',

            description: "Hi Magdalena, I've seen your challenge...",

            time: '2 m ago',

            hasAction: true,

            actionText: 'Reply the message',

          ),

          _buildNotificationItem(

            icon: Icons.flash_on,

            title: 'Ramadhan Flash Sale!',

            description: 'Get 20% discount for first transaction in this month!😍😍',

            time: '2 m ago',

          ),

          _buildNotificationItem(

            icon: Icons.emoji_events,

            title: 'Challange Completed',

            description: 'Congratulation! you have completed course by Jerremy Mamika',

            time: '10 m ago',

          ),

        ],

      ),

    );

  }



  Widget _buildNotificationItem({

    required IconData icon,

    required String title,

    required String description,

    required String time,

    bool hasAction = false,

    String actionText = '',

  }) {

    return ListTile(

      leading: CircleAvatar(

        backgroundColor: Colors.grey[200],

        child: Icon(icon, color: Colors.black),

      ),

      title: Text(title, style: TextStyle(fontWeight: FontWeight.bold)),

      subtitle: Column(

        crossAxisAlignment: CrossAxisAlignment.start,

        children: [

          Text(description),

          if (hasAction)

            TextButton(

              onPressed: () {

                // TODO: Implement action

              },

              child: Text(actionText, style: TextStyle(color: Colors.purple)),

            ),

        ],

      ),

      trailing: Text(time, style: TextStyle(color: Colors.grey)),

      contentPadding: EdgeInsets.symmetric(horizontal: 16, vertical: 8),

    );

  }

}